doc/tutorials/dockerd-specification.md

```diff
@@ -0,0 +1,429 @@
+# How to Write a Specification in Elektra for dockerd
+
+## Overview
+
+### Introduction
+
+In this tutorial you will learn how to interactively use the `SpecElektra` specification
+language and `kdb` to write a configuration specification for [dockerd](https://docs.docker.com/engine/reference/commandline/dockerd/).
+
+### What you should already know
+
+- Already know how to write a [specification](specification.md)
+
+### What you’ll learn
+
+- how to create and mount a specification using `kdb`
+- how to add keys with different types, defaults, enums, array specifications, wildcard specifications and examples to your specification and how to validate them
+
+### What you'll do
+
+- use `kdb` to create and mount a specification for [dockerd](https://docs.docker.com/engine/reference/commandline/dockerd/)
+- define defaults, array / wildcard specifications, examples and checks for keys in the validation
+- use the specification as a starting point for customizing the configuration of installed applications
+
+### Scope
+
+In this tutorial we will introduce a possible specification for [dockerd](https://docs.docker.com/engine/reference/commandline/dockerd/).
+Using `kdb` we will configure the specification.
+
+> NOTE: As the specification for dockerd is quite big we will only present a sample for the above mentioned metakeys and link to a full specification [dockerd-full-spec](../../examples/spec/dockerd.ini).
+
+## Getting Started
+
+Before we start just an overview of the structure:
+
+- Specification file location: `~/.config/dockerd/dockerd-daemon.ni`
+- Parent specification key: `spec:/sw/dockerd/dockerd/#0/current`
+
+## Specification Types (values)
+
+`Elektra` supports multiple types which leads to a more flexible specification.
+See [type plugin](../../src/plugins/type/README.md) for information about all the types that are supported.
+
+## Mount Setup
+
+### Step 1: Mount dockerd specification
+
+First you need to mount a specification file, in this case `dockerd-daemon.ni` to the `spec:/` namespace.
+You can define the path inside the `spec:/` namespace as `/sw/docker/dockerd/#0/current`, refer to
+[the documentation](https://www.libelektra.org/tutorials/integration-of-your-c-application) to find out more about constructing the name.
+
+```sh
+sudo kdb mount "$PWD/.config/dockerd/dockerd-daemon.ni" spec:/sw/docker/dockerd/#0/current ni
```

Comment:
If you use $PWD then directly refer to the correct file within Elektra's folder:

```suggestion
sudo kdb mount "$PWD/examples/spec/dockerd.ini" spec:/sw/docker/dockerd/#0/current ni
```

Reply:
oh, ok, thanks!